qa/rpc-tests/excessive.py

```diff
@@ -12,17 +12,231 @@
+          assert_equal(mbefore[1:],[(0, 0), (0, 0), (0, 0)])  # verify that the transaction did not propagate
+          assert(mbefore[0][0] > 0) # verify that the transaction is in my node
+
+          logging.info("Test a large transaction in block < 1MB")
+          largeBlock = self.nodes[0].generate(1)
+          self.sync_blocks()
+          counts = [ x.getblockcount() for x in self.nodes ]
+          latest = counts[0]
+          assert_equal(counts, [latest,latest,latest,latest]) # Verify that all nodes accepted the block, even if some of them didn't have the transaction.  They should all accept a <= 1MB block with a tx <= 1MB
+
+  	  # mafter = [ (lambda y: (y["size"],y["bytes"]))(x.getmempoolinfo()) for x in self.nodes]
+
+        if self.extended:  # this test checks the behavior of > 1MB blocks with excessive transactions.  it takes a LONG time to generate and propagate 1MB+ txs.
+  	  # Create a LOT of UTXOs for the next test
+          wallet = self.nodes[0].listunspent()
+          wallet.sort(key=lambda x: x["amount"],reverse=True)
+          logging.info("Create lots of UTXOs...")
+          n=0
+
+          for w in wallet:
+            split_transaction(self.nodes[0], [w], addrs[n:100+n])
+            n+=100
+            if n >= len(addrs): n=0
+	
+          self.nodes[0].generate(1)
+	  self.sync_all()
+
+          logging.info("Building > 1MB block...")
+          self.nodes[0].set("net.excessiveTxn=1000000")  # Set the excessive transaction size larger for this node so we can generate an "excessive" block for the other nodes
+        
+          wallet = self.nodes[0].listunspent()
+          wallet.sort(key=lambda x: x["amount"],reverse=False)
+
+          # Generate 1 MB worth of transactions        
+          size = 0
+          count = 0
+          while size < 1000000:
+            count+=1
+            utxo = wallet.pop()
+            outp = {}
+	    outp[addrs[count%len(addrs)]] = utxo["amount"]
+            txn = self.nodes[0].createrawtransaction([utxo], outp)
+            signedtxn = self.nodes[0].signrawtransaction(txn)
+            size += len(binascii.unhexlify(signedtxn["hex"]))
+            self.nodes[0].sendrawtransaction(signedtxn["hex"])
+
+          # Now generate a > 100kb transaction & mine it into a > 1MB block
+
+          self.nodes[0].setminingmaxblock(2000000)
+          wallet.sort(key=lambda x: x["amount"],reverse=True)
+	  (tx, vin, vout, txid) = split_transaction(self.nodes[0],wallet[0:2000],[addrs[0]],txfeePer=60)
+          logging.debug("Transaction Length is: ", len(binascii.unhexlify(tx)))
+          # assert(binascii.unhexlify(tx) > 100000) # txn has to be big for the test to work
+	  origCounts = [ x.getblockcount() for x in self.nodes ]
+          mpool = [ (lambda y: (y["size"],y["bytes"]))(x.getmempoolinfo()) for x in self.nodes]
+          print mpool
+          largeBlock = self.nodes[0].generate(1)
+          time.sleep(10) # can't sync b/c nodes won't be in sync
```

Comment:
Fixed sleeps are bad news for tests under most circumstances.
I see this one repeats a couple of times further down as well.
Is there a way this can be made into a while loop that checks for a successful sync condition, like you did elsewhere in the script?
This would be much more robust and give the tester a much better idea of what went wrong if the test gets stuck on syncing rather than failing due to an elapsed timeout (which could have a myriad reasons).